SIS_import/sis_upload.rb

```diff
@@ -58,12 +59,32 @@
 # the path of Canvas API call
 API_PATH="/api/v1/"
 
+
+# variables for Canva API call throttling
+@Canvas_call_count = 0
+@Canvas_start_time = Time.now
+@Canvas_time_interval_in_seconds = 3600
+@Canvas_end_time = @Canvas_start_time + @Canvas_time_interval_in_seconds # one minute apart
+@Canvas_allowed_call_number_during_interval = 3000
+
+# make sure the Canvas call is within API usage quota
+def Canvas_API_CALL_check
+	call_hash = sleep_according_to_timer_and_api_call_limit(@Canvas_start_time, @Canvas_end_time, @Canvas_call_count, @Canvas_time_interval_in_seconds, @Canvas_allowed_call_number_during_interval)
+	@Canvas_start_time = call_hash["start_time"]
+	@Canvas_end_time = call_hash["end_time"]
+	@Canvas_call_count = call_hash["call_count"]
+	@Canvas_call_count = @Canvas_call_count+1
+end
+
 # Ruby URI.escape has been deprecated.
 # Addressable::URI.escape seems to be a viable solution, which offers url encoding, form encoding and normalizes URLs.
 # http://stackoverflow.com/questions/2824126/whats-the-difference-between-uri-escape-and-cgi-escape
 
 ## make Canvas API GET call
 def Canvas_API_GET(url)
+	# make sure the Canvas call is within API usage quota
+	Canvas_API_CALL_check()
+	p "Canvas API GET #{url}"
```

Comment:
Should use Ruby logging instead of print statements.  If it looks hard to do let's talk.